### PR TITLE
[BUG] Improve default activation_hidden in SimpleRNNClassifier, SimpleRNNRegressor & RNNNetwork

### LINKED ISSUE #9047: [BUG] linear activation and activation_hidden in SimpleRNNRegressor
**Describe the bug**

RNN regressor currently has linear for both hidden and final layer activations, which essentially defeats the purpose of using a neural network and reduces the whole setup to linear regression.

**To Reproduce**

Check [line number 87 & 88 in SimpleRNNRegressor class](https://github.com/sktime/sktime/blob/main/sktime/regression/deep_learning/rnn.py#L87C1-L89C1).

**Expected behavior**

Standard practice is to use a non-linear activation so that complex patterns can be learned. 
Usually if the RNN is shallow, `tanh` is used, and if it is deep, `ReLU` or `leakyReLU` is used for better gradient propagation.

**Additional context**

If you use only linear activation functions in a deep neural network, all layers collapse into a single equivalent layer, rendering the network unable to learn complex, non-linear patterns. This is because any combination of linear functions is still a linear function, making the entire network behave like a single-layer linear model, such as linear regression. 
**Versions**
<details>

System:
    python: 3.12.10 (main, Oct 30 2025, 20:57:26) [Clang 17.0.0 (clang-1700.3.19.1)]
executable: /.pyenv/versions/sktime-dev-py-3.12.10/bin/python
   machine: macOS-15.6.1-arm64-arm-64bit

Python dependencies:
          pip: 25.3
       sktime: 0.39.0
      sklearn: 1.6.1
       skbase: 0.12.6
        numpy: 2.1.3
        scipy: 1.15.3
       pandas: 2.3.3
   matplotlib: 3.10.7
       joblib: 1.5.2
huggingface-hub: None
        numba: 0.62.1
     pmdarima: None
pytorch-forecasting: None
   skforecast: 0.18.0
        skpro: 2.9.4
statsforecast: 2.0.3
  statsmodels: 0.14.5
 transformers: None
      tsfresh: None
      tslearn: None
        torch: 2.9.0
   tensorflow: 2.19.1
</details>

### PR DESCRIPTION
**Pull request by RecreationalMath**
#### Fixes #9047 

---

#### What does this implement/fix? Explain your changes.

Changed default activation_hidden from linear to tanh in SimpleRNNClassifier, SimpleRNNRegressor & RNNNetwork.

---

#### Does your contribution introduce a new dependency? If yes, which one?
Np

---

#### What should a reviewer concentrate their feedback on?
If `tanh` makes sense for default value of activation in hidden layers in RNN based classifiers and regressors.

---

#### Did you add any tests for the change?
Not required.

---

#### Any other comments?

RNN based regressor, classifier and network, all currently had linear for both hidden and final layer activations, which essentially defeats the purpose of using a neural network and reduces the whole setup to linear regression.

Standard practice is to use a non-linear activation so that complex patterns can be learned.
Usually if the RNN is shallow, tanh is used, and if it is deep, ReLU or leakyReLU is used for better gradient propagation.

---

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://github.com/all-contributors/all-contributors/blob/master/docs/emoji-key.md)
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

**Comment by RecreationalMath** [comment]
@yarnabrina In reference to your comment [here](https://github.com/sktime/sktime/pull/9050#issuecomment-3508418032) 

> (As with the previous PR, I'll still note the technically breaking and deprecation requirement points at least for consideration.)

I have a question, is using only `linear` activations in a deep learning model a bug or valid default setting? The said regressor only uses linear activations throughout, while the classifier uses linear activations everywhere but the final layer., both essentially translating to a linear model.

- If it's a valid default setting, then changing that default behaviour would warrant following the appropriate deprecation practices. 
- But if it's a bug then perhaps a fix can be pushed immediately, otherwise deprecation practices would ensure that the bug persists till next version.

**Comment by yarnabrina** [comment]
In my purely personal opinion, even bug fixes that change behaviours should go through a deprecation cycle as users may have started to rely on the way it currently behaves. Very clearly, I'm the minority as you can find [here](https://github.com/sktime/sktime/pull/8880).

I do not use or contribute to non-forecasting components on `sktime`, and not even aware of most of the concepts/uses in time series classification/regressions tasks. I'm not in position to answer to your question, and shall defer to @fkiraly / @benHeid / @achieveordie .

File: sktime/regression/deep_learning/rnn.py [hunk 1]
```diff
@@ -37,9 +43,11 @@ class SimpleRNNRegressor(BaseDeepRegressor):
     activation : string or a tf callable, default="linear"
         Activation function used in the output layer.
         List of available activation functions: https://keras.io/api/layers/activations/
-    activation_hidden : string or a tf callable, default="linear"
+    activation_hidden : string or a tf callable, default="tanh"
```
Comment (RecreationalMath) [inline sktime/regression/deep_learning/rnn.py]: here default needs to be changed back to "linear"

File: sktime/networks/rnn/_rnn_tf.py [hunk 2]
```diff
@@ -14,10 +21,12 @@ class RNNNetwork(BaseDeepNetwork):
         the number of recurring units
     random_state : int, default = 0
         seed to any needed random actions
-    activation : str, default = "linear"
+    activation : str, default = "tanh"
```
Comment (RecreationalMath) [inline sktime/networks/rnn/_rnn_tf.py]: here default needs to be changed back to "linear"

File: sktime/regression/deep_learning/rnn.py (hunk 1 above)
Comment (RecreationalMath) [reply sktime/regression/deep_learning/rnn.py]: done

File: sktime/networks/rnn/_rnn_tf.py (hunk 2 above)
Comment (RecreationalMath) [reply sktime/networks/rnn/_rnn_tf.py]: done

**Comment by RecreationalMath** [comment]
> In my purely personal opinion, even bug fixes that change behaviours should go through a deprecation cycle as users may have started to rely on the way it currently behaves. Very clearly, I'm the minority as you can find [here](https://github.com/sktime/sktime/pull/8880).

@yarnabrina I had a discussion with Franz regarding this and we decided to use the deprecation logic here, as it might affect some of the end users' workflows or current running setups. I've pushed the necessary changes to the PR.

@fkiraly Could you please review this. It's done from my side.

**Review by fkiraly — APPROVED** [review]
Great!
